src/main/java/org/semux/consensus/SemuxSync.java

```diff
@@ -315,16 +336,61 @@ private void process() {
             return; // This is important because stop() only notify
         }
 
+        // Perform fast sync only if all blocks in current validator sets have been
+        // forged
+        synchronized (lock) {
+            if (!fastSync) {
+                // fastSync value is updated at the beginning of each set
+                if (latest % config.getValidatorUpdateInterval() == 0) {
+                    if (target.get() >= latest + config.getValidatorUpdateInterval()) {
+                        toFinalize.clear();
+                        currentSet.clear();
+                        lastBlockInSet = latest + config.getValidatorUpdateInterval();
+                        fastSync = true;
+                    }
+                }
```

Comment:
I'm concerned that if more and more people are using `fastsync`, votes for blocks inside an interval are less-likely validated. Eventually, we may never be able to sync the blocks in `[start_block, last_block_of_interval]`